include/psim/core/configuration.hpp

```diff
@@ -56,21 +56,17 @@ class Configuration {
 
  protected:
   /** @brief Map containing the parameter pointers.
-   *
-   *  https://stackoverflow.com/questions/9139748/using-stdreference-wrapper-as-the-key-in-a-stdmap
    */
-  std::unordered_map<std::reference_wrapper<std::string const>,
-                     std::unique_ptr<ParameterBase const>,
-                     std::hash<std::string>, std::equal_to<std::string>>
-                     _parameters;
+  std::unordered_map<std::string, std::unique_ptr<ParameterBase const>>
```

Comment:
so this was the critical change related to the bug?

Reply:
Yes this was absolutely critical. I'm going to add a better explanation in the description of exactly what was going on.